src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -60,44 +66,80 @@ public JSONGenerator(File templateDir, File outputDir) {
+
+		RequiredPlugins requirements = syntaxClass.getAnnotation(RequiredPlugins.class);
+		syntaxJsonObject.add("requirements", convertToJsonArray(requirements.value()));
 
-		syntaxJsonObject.add("patterns", convertToJsonArray(syntaxInfo.getPatterns()));
 		return syntaxJsonObject;
 	}
 
 	/**
 	 * Generates the documentation JsonObject for an event
-	 * @param eventInfo the event to generate the documentation object for
+	 * @param info the event to generate the documentation object for
 	 * @return a documentation JsonObject for the event
 	 */
-	private JsonObject generateEventElement(SkriptEventInfo<?> eventInfo) {
+	private JsonObject generateEventElement(SkriptEventInfo<?> info) {
 		JsonObject syntaxJsonObject = new JsonObject();
-		syntaxJsonObject.addProperty("id", DocumentationIdProvider.getId(eventInfo));
-		syntaxJsonObject.addProperty("name", eventInfo.name);
-		syntaxJsonObject.addProperty("since", eventInfo.getSince());
-		syntaxJsonObject.add("description", convertToJsonArray(eventInfo.getDescription()));
-		syntaxJsonObject.add("examples", convertToJsonArray(eventInfo.getExamples()));
-		syntaxJsonObject.add("patterns", convertToJsonArray(eventInfo.patterns));
+		syntaxJsonObject.addProperty("id", DocumentationIdProvider.getId(info));
+		syntaxJsonObject.addProperty("name", info.getName());
+		syntaxJsonObject.addProperty("since", info.getSince());
+
+		syntaxJsonObject.add("patterns", convertToJsonArray(info.getPatterns()));
+		syntaxJsonObject.add("description", convertToJsonArray(info.getDescription()));
+		syntaxJsonObject.add("requirements", convertToJsonArray(info.getRequiredPlugins()));
+		syntaxJsonObject.add("examples", convertToJsonArray(info.getExamples()));
+
+
+		boolean cancellable = false;
+		for (Class<? extends Event> event : info.events) {
+			if (Cancellable.class.isAssignableFrom(event) || BlockCanBuildEvent.class.isAssignableFrom(event)) {
+				cancellable = true;
+				break;
+			}
+		}
+
+		syntaxJsonObject.addProperty("cancellable", cancellable);
+
+		Set<String> eventValues = new TreeSet<>();
+		Multimap<Class<? extends Event>, EventValueInfo<?, ?>> allEventValues = EventValues.getPerEventEventValues();
+		for (Class<? extends Event> supportedEvent : info.events) {
+			for (Class<? extends Event> event : allEventValues.keySet()) {
+				if (!event.isAssignableFrom(supportedEvent))
+					continue;
+
+				Collection<EventValueInfo<?, ?>> eventValueInfos = allEventValues.get(event);
+
+				for (EventValueInfo<?, ?> eventValueInfo : eventValueInfos) {
+					Class<?>[] excludes = eventValueInfo.excludes();
+					if (excludes != null && Set.of(excludes).contains(event))
+						continue;
+
+					ClassInfo<?> exactClassInfo = Classes.getExactClassInfo(eventValueInfo.c());
+					if (exactClassInfo == null)
+						continue;
+
+					eventValues.add("event-%s".formatted(exactClassInfo.getName()));
+				}
+			}
+		}
+		syntaxJsonObject.addProperty("event-values", Joiner.on(", ").join(eventValues));
```

Comment:
can we use camel case please?

Reply:
also, I think it would be more valuable if we provided the values separately and then the best way to display the values up to the consumer. we could potentially do the object idea I mentioned above too :)